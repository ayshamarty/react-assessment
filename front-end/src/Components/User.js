import React from "react";

import ListGroup from "react-bootstrap/ListGroup";

function User(props) {

      return (
    <div>
      <ListGroup.Item
      >
        {props.user}
      </ListGroup.Item>
    </div>
  );

}

export default User;