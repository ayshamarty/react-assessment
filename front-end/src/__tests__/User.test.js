import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import User from "../Components/User";

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<User />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders shallow without crashing", () => {
  shallow(<User />);
});

it('snapshot User', () => {
  const wrapper = renderer.create(<User/>).toJSON()
  expect(wrapper).toMatchSnapshot();
})

