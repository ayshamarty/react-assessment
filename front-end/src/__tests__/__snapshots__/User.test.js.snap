// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot User 1`] = `
<div>
  <div
    className="list-group-item"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
